torch/compiler/__init__.py

```diff
@@ -92,3 +101,64 @@ def disable(fn=None, recursive=True):
     import torch._dynamo
 
     return torch._dynamo.disable(fn, recursive)
+
+
+def dynamic_dim(t: torch.Tensor, index: int):
+    """
+    Note - [On Export Dynamic Dimension UX]
```

Comment:
I'm not sure this will render correctly so worth taking a look at docbuild and seeing what this will actually look like